src/mains/CMakeLists.txt

```
@@ -22,6 +28,19 @@ else()
                           SOURCES orcamodelHofX3D.cc
                           LIBS    orcamodel
                         )
+
+  ecbuild_add_executable( TARGET  orcamodel_3DVar.x
+                          SOURCES orcamodel3DVar.cc
+                          LIBS    orcamodel
+                        )
 endif()
+
+ecbuild_add_executable( TARGET  orcamodel_ErrorCovarianceToolbox.x
+                        SOURCES orcamodelErrorCovarianceToolbox.cc
+                        LIBS    orcamodel
+                      )
+
 oops_output_json_schema( "orcamodel_hofx.x" )
 oops_output_json_schema( "orcamodel_hofx3D.x" )
+#oops_output_json_schema( "orcamodel_3DVar.x" )
+#oops_output_json_schema( "orcamodel_ErrorCovarianceToolbox.x" )
```

```suggestion
oops_output_json_schema( "orcamodel_3DVar.x" )
oops_output_json_schema( "orcamodel_ErrorCovarianceToolbox.x" )
```

Do these work? It would be great to get them working as it means you get all the schema validation for free in your editor when you are writing configurations.

---

Unfortunately it fails on the build step. It's not clear from the error what the issue actually is. Here's what it say for the orcamodel_ErrorCovarianceToolbox.x

[527/534] Linking CXX executable bin/orcamodel_ErrorCovarianceToolbox.x
[528/534] cd /data/users/frld/jedi/build-mo-bundle/etc && /data/users/frld/jedi/build-mo-bundle/bin/orcamodel_ErrorCovarianceToolbox.x --output-json-schema=orcamodel_ErrorCovarianceToolbox.x.schema.json
Configured to load plugins []
Plugins manifest candidate paths [~eckit/share/plugins,~/share/plugins]
Loading library paths from /spice/data/users/frld/jedi/build-mo-bundle//etc/paths
Failed to read /spice/data/users/frld/jedi/build-mo-bundle//etc/paths --  (No such file or directory)
Scanning for plugins manifest path ~eckit/share/plugins resolved to /spice/data/users/darth/opt/mobbs-55/base/spice_gnu/share/plugins
Found plugin manifest /spice/data/users/darth/opt/mobbs-55/base/spice_gnu/share/plugins/atlas-orca.yml
Loaded plugin manifest LocalConfiguration[root={name => atlas-orca , namespace => int.ecmwf , url => https://sites.ecmwf.int/docs/atlas , version => 0.3.1 , library => atlas-orca}]
Going to load following plugins [int.ecmwf.atlas-orca]
Plugin [atlas-orca] already loaded
Initializing plugin [atlas-orca]
Parsing Lib atlas-orca config file /spice/data/users/darth/opt/mobbs-55/base/spice_gnu/etc/atlas-orca/config.yaml
Lib atlas-orca configuration: YAMLConfiguration[path=/spice/data/users/darth/opt/mobbs-55/base/spice_gnu/etc/atlas-orca/config.yaml, root={atlas-orca-grids-path => ~atlas-orca/share/atlas/grids/orca/grids.yaml , atlas-orca-caching => false , atlas-orca-validate-uid => false , atlas-orca-compute-uid => false}]
Application orcamodel_ErrorCovarianceToolbox loaded libraries: [atlas,atlas-orca,eckit,fckit,oops]
Parsing Lib eckit config file /spice/data/users/darth/opt/mobbs-55/base/spice_gnu/etc/eckit/config.yaml
Lib eckit configuration: YAMLConfiguration[path=<string>, root=(nil)]
Output JSON Schema file: orcamodel_ErrorCovarianceToolbox.x.schema.json

---

I was trying to find other examples of oops_output_json_schema in the mo-bundle. Lfric-jedi don't have any, but the QG model does. See oops/qg/mains/CMakeLists.txt I note that some of oops_output_json_schema are commented out with #TODO . It's not clear why but perhaps something isn't working yet at a lower level than I'm working on.

---

Hmm OK that is a shame. I think comment it out for now, with a comment above referencing qg. Something we can look into later perhaps.